src/group/libp2p_group_relcast_server.erl

```diff
@@ -202,20 +198,17 @@ handle_cast({send_result, {_Key, _Index}, {error, _Error}}, State=#state{self_in
     %% For any other result error response we leave the worker busy
     %% and we wait for it to send us a new ready on a reconnect.
     {noreply, State};
-handle_cast({handle_ack, Index}, State=#state{self_index=_SelfIndex}) ->
+handle_cast({handle_ack, Index, Seq}, State=#state{self_index=_SelfIndex}) ->
     %% Received when a previous message had a send_result of defer.
     %% We don't handle another defer here so it falls through to an
     %% unhandled cast below.
-    case lookup_worker(Index, State) of
-        #worker{msg_key=MsgKey} when is_reference(MsgKey) ->
-            %% Delete the outbound message for the given index
-            {ok, NewRelcast} = relcast:ack(Index, MsgKey, State#state.store),
-            {noreply, dispatch_next_messages(ready_worker(Index, undefined, State#state{store=NewRelcast}))};
-        _ ->
-            lager:debug("Unexpected ack for ~p", [Index]),
-            {noreply, dispatch_next_messages(State)}
-    end;
-handle_cast({handle_data, Index, Msg}, State=#state{self_index=_SelfIndex}) ->
+    %% Delete the outbound message for the given index
+    {ok, NewRelcast} = relcast:ack(Index, Seq, State#state.store),
+    Worker = lookup_worker(Index, State),
+    InFlight = relcast:in_flight(Index, NewRelcast),
```

Comment:
I notice InFlight being updated every time right before calling `update_worker`. can we move fetching inflight there?